src/e2sz/e2sz.c

```diff
@@ -0,0 +1,39 @@
+// Finds the size of an ext4 filesystem located at the beginning of a given device.
+
+#include <inttypes.h>
+#include <error.h>
+#include <ext2fs/ext2fs.h>
+
+static void usage() {
+	printf("Usage: e2sz <device>\n");
+}
+
+int main(int argc, char *argv[]) {
+	ext2_filsys fs;
+	errcode_t err;
+
+	uint64_t fs_size = 0;
+	int retc = 1;
+
+
+	if (argc != 2) {
+		usage();
+		goto out;
+	}
+
+	err = ext2fs_open(argv[1], 0, 0, 0, unix_io_manager, &fs);
+
+	if (err != 0) {
+		printf("%s\n", strerror(errno));
+		goto out;
+	}
+
+	fs_size = ext2fs_blocks_count(fs->super) * fs->blocksize;
+
+	printf("%" PRIu64 "\n", fs_size);
+
+	retc = 0;
+
+out:
```

Comment:
shouldn't there be a close or other cleanup here? If there isn't any cleanup then drop the goto and just return 1